src/accelerate/accelerator.py

```diff
@@ -3045,19 +3045,23 @@ def free_memory(self):
         >>> accelerator = Accelerator()
         >>> model, optimizer, scheduler = ...
         >>> model, optimizer, scheduler = accelerator.prepare(model, optimizer, scheduler)
-        >>> accelerator.free_memory()
-        >>> del model, optimizer, scheduler
+        >>> model, optimizer, scheduler = accelerator.free_memory(model, optimizer, scheduler)
         ```
         """
+        objects = release_memory(*objects)
         self._schedulers = []
         self._optimizers = []
         self._models = []
         self._dataloaders = []
-        self.deepspeed_engine_wrapped = None
+        # Deepspeed needs a bit more prep
+        if hasattr(self, "deepspeed_engine_wrapped"):
+            if self.deepspeed_engine_wrapped is not None:
```

Comment:
`deepspeed_engine_wrapped` doesn't have destroy method, it should be `deepspeed_engine_wrapped.engine.destroy()`. Also, it would be better to handle DeepSpeed engine before the statement `objects = release_memory(*objects)`.

Reply:
@pacman100 the trainer does already :) 

https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1885

Reply:
It is called at the start of training, I meant that people would like to call this method as per their convenience after training.

Reply:
Ahhh.

I'll think about how we can go about it, since the optimizer is baked into the trainer so the user really just needs to pass in the `model` I think

Reply:
Done, please see https://github.com/huggingface/transformers/pull/30549